core/src/org/apache/cloudstack/agent/transport/AnswerArrayTypeAdaptor.java

```diff
@@ -0,0 +1,33 @@
+package org.apache.cloudstack.agent.transport;
+
+import java.io.IOException;
+import java.lang.reflect.Array;
+
+import com.cloud.agent.api.Answer;
+import com.cloud.utils.exception.CloudRuntimeException;
+import com.google.gson.stream.JsonWriter;
+
+public class AnswerArrayTypeAdaptor extends GenericArrayTypeAdaptor<Answer> {
+
+    @Override
+    protected Answer[] newArray(int size) {
+        Answer[] answers = (Answer[])Array.newInstance(Answer.class, size);
```

Comment:
it is the same and the optimizer will treat it the same but this is easier to debug